addons/web/static/src/views/form/form_error_dialog/form_error_dialog.js

```diff
@@ -20,47 +20,39 @@ export class FormErrorDialog extends Component {
         await this.props.onDiscard();
         this.props.close();
     }
+
+    async stay() {
+        await this.props.onStayHere();
+        this.props.close();
+    }
 }
 FormErrorDialog.template = "web.FormErrorDialog";
 FormErrorDialog.components = { Dialog };
 
-function makeFormErrorHandler(onDiscard) {
-    return (env, error, originalError) => {
-        if (
-            originalError &&
-            originalError.legacy &&
-            originalError.message &&
-            originalError.message instanceof RPCError
-        ) {
-            const event = originalError.event;
-            originalError = originalError.message;
-            error.unhandledRejectionEvent.preventDefault();
-            if (event.isDefaultPrevented()) {
-                // in theory, here, event was already handled
-                return true;
-            }
-            event.preventDefault();
-
-            env.services.dialog.add(FormErrorDialog, {
-                message: originalError.message,
-                data: originalError.data,
-                onDiscard,
-            });
-
+function formSaveErrorHandler(env, error, originalError) {
+    if (originalError.__triggeredOnFormSave) {
+        const event = originalError.event;
+        error.unhandledRejectionEvent.preventDefault();
+        if (event.isDefaultPrevented()) {
+            // in theory, here, event was already handled
             return true;
         }
-        return false;
-    };
-}
-
-let formId = 0;
+        event.preventDefault();
+
+        env.services.dialog.add(
+            FormErrorDialog,
+            {
+                message: originalError.message.message,
+                data: originalError.message.data,
+                onDiscard: originalError.onDiscard,
+                onStayHere: originalError.onStayHere,
+            },
+            {
+                onClose: originalError.onStayHere, // crap
```

Comment:
remove comment

Reply:
also, ask @bouvyd about closing with ESC

Reply:
> also, ask @bouvyd about closing with ESC

For the record: closing on ESC indeed, with behaviour equal to "STAY HERE" (but discussed yesterday in DMs)